### PR TITLE
core: Add uuid() interpolate function.

### PR DESCRIPTION
**Pull request by phinze**
Utilize hashicorp's go-uuid library for proper random seeding setup.

### CODE REVIEW

**Comment by apparentlymart** [comment]
Is this building on #3563 or is it a fresh implementation?

**Comment by phinze** [comment]
(Working on the dep vendoring now too.)

File: website/source/docs/configuration/resources.html.md [hunk 1]
```diff
@@ -68,6 +68,8 @@ The `lifecycle` block allows the following keys to be set:
       destruction of a given resource. When this is set to `true`, any plan
       that includes a destroy of this resource will return an error message.
 
+<a id="ignore-changes"></a>
```
Comment (jen20) [inline website/source/docs/configuration/resources.html.md]: This seems stray for this PR but still desirable to add :+1:

**Comment by jen20** [comment]
This looks good to me. :+1:

**Comment by gjohnson** [comment]
:+1:

**Comment by gozer** [comment]
That's great, but I was hoping something like this would become posible:

```
resource "aws_s3_bucket" "test" {
    lifecycle {
      ignore_changes = ["bucket"]
    }

    bucket = "some-prefix-${uuid()}"
    acl = "private"
}
```
However, the bucket creation itself initially fails, am I doing something obviously wrong here?

**Comment by phinze** [comment]
@gozer you're doing it right, that's a bug with ignore_changes that i'm actively working on fixing! stay tuned

**Comment by gozer** [comment]
@phinze great, thanks for the feedback. Do you have an issue number for that ignore_change bug?

**Comment by phinze** [comment]
@gozer just opened a roll-up issue for my efforts :point_right: #5627

**Pull request merged**